### PR TITLE
Dssp and hover name

### LINKED ISSUE #171: local pdb + dssp = error
**Describe the bug**
When using a local pdb file and a ProteinGraphConfig that contains a DSSPConfig, DSSPConfig always tries to download the pdb file, even if it is in the pdb_dir, and throws an error if the name of the pdb file is not a PDB ID.

**To Reproduce**

```
from functools import partial
from graphein.protein.subgraphs import extract_subgraph_from_chains
from graphein.protein.config import ProteinGraphConfig, DSSPConfig
from graphein.protein.features.nodes.amino_acid import expasy_protein_scale, meiler_embedding
from graphein.protein.features.nodes import asa, rsa
from graphein.protein.edges.distance import (add_peptide_bonds,
                                             add_hydrogen_bond_interactions,
                                             add_disulfide_interactions,
                                             add_ionic_interactions,
                                             add_aromatic_interactions,
                                             add_aromatic_sulphur_interactions,
                                             add_cation_pi_interactions
                                            )


conf_functions = {"edge_construction_functions": [add_peptide_bonds,
                                                  add_aromatic_interactions,
                                                  add_hydrogen_bond_interactions,
                                                  add_disulfide_interactions,
                                                  add_ionic_interactions,
                                                  add_aromatic_sulphur_interactions,
                                                  add_cation_pi_interactions],
                  "graph_metadata_functions": [asa, rsa],                                        # Add ASA and RSA features.
                  "node_metadata_functions": [meiler_embedding,partial(expasy_protein_scale, add_separate=True)], # Add expasy features (partial: each feature is added under a separate key)
                  "dssp_config":DSSPConfig(),                                                    # Add DSSP config in order to compute ASA and RSA.
                  "pdb_dir": '/vol/tmp/kormanav/pdb_dir'
                 }        
batch_config = ProteinGraphConfig(**conf_functions)

construct_graph(config=batch_config, pdb_path="/vol/tmp/kormanav/6rew_copy.pdb")
```
Results in:
```
DEBUG:graphein.protein.graphs:Deprotonating protein. This removes H atoms from the pdb_df dataframe
DEBUG:graphein.protein.graphs:Detected 1365 total nodes
INFO:graphein.protein.edges.distance:Found: 234 aromatic-aromatic interactions
INFO:graphein.protein.edges.distance:Found 532 hbond interactions.
INFO:graphein.protein.edges.distance:Found 55 hbond interactions.
DEBUG:graphein.protein.edges.distance:0 CYS residues found. Cannot add disulfide interactions with fewer than two CYS residues.
INFO:graphein.protein.edges.distance:Found 11848 ionic interactions.
Downloading PDB structure '6rew_copy'...
ERROR:graphein.protein.utils:PDB file 6rew_copy not found and no replacement                       structure found in obsolete lookup.
Desired structure doesn't exists
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Input In [20], in <cell line: 39>()
     37 graph_list = []
     38 y_list = []
---> 39 construct_graph(config=batch_config, pdb_path="/vol/tmp/kormanav/6rew_copy.pdb")
     40 """for idx, pdb_p in enumerate(tqdm(pdb_paths)):
     41     print('pdb:', pdb_p)
     42     try:
   (...)
     46         print(f'PDB #{idx}: processing error!')
     47         pass"""

File ~/repos_dev/graphein/graphein/protein/graphs.py:614, in construct_graph(config, pdb_path, pdb_code, chain_selection, df_processing_funcs, edge_construction_funcs, edge_annotation_funcs, node_annotation_funcs, graph_annotation_funcs)
    612 # Annotate additional graph metadata
    613 if config.graph_metadata_functions is not None:
--> 614     g = annotate_graph_metadata(g, config.graph_metadata_functions)
    616 # Annotate additional edge metadata
    617 if config.edge_metadata_functions is not None:

File ~/repos_dev/graphein/graphein/utils/utils.py:69, in annotate_graph_metadata(G, funcs)
     58 """
     59 Annotates graph with graph-level metadata
     60 
   (...)
     66 :rtype: nx.Graph
     67 """
     68 for func in funcs:
---> 69     func(G)
     70 return G

File ~/repos_dev/graphein/graphein/protein/features/nodes/dssp.py:239, in asa(G)
    230 def asa(G: nx.Graph) -> nx.Graph:
    231     """
    232     Adds ASA of each residue in protein graph as calculated by DSSP.
    233 
   (...)
    237     :rtype: nx.Graph
    238     """
--> 239     return add_dssp_feature(G, "asa")

File ~/repos_dev/graphein/graphein/protein/features/nodes/dssp.py:174, in add_dssp_feature(G, feature)
    144 """
    145 Adds add_dssp_feature specified amino acid feature as calculated
    146 by DSSP to every node in a protein graph
   (...)
    171 :rtype: nx.Graph
    172 """
    173 if "dssp_df" not in G.graph:
--> 174     G = add_dssp_df(G, G.graph["config"].dssp_config)
    176 config = G.graph["config"]
    177 dssp_df = G.graph["dssp_df"]

File ~/repos_dev/graphein/graphein/protein/features/nodes/dssp.py:122, in add_dssp_df(G, dssp_config)
    120 # Check for existence of pdb file. If not, download it.
    121 if not os.path.isfile(config.pdb_dir / pdb_id):
--> 122     pdb_file = download_pdb(config, pdb_id)
    123 else:
    124     pdb_file = config.pdb_dir + pdb_id + ".pdb"

File ~/repos_dev/graphein/graphein/protein/utils.py:100, in download_pdb(config, pdb_code)
     95         log.error(
     96             f"PDB file {pdb_code} not found and no replacement \
     97                   structure found in obsolete lookup."
     98         )
     99 # Rename file to .pdb from .ent
--> 100 os.rename(
    101     config.pdb_dir / f"pdb{pdb_code}.ent",
    102     config.pdb_dir / f"{pdb_code}.pdb",
    103 )
    105 # Assert file has been downloaded
    106 assert any(pdb_code in s for s in os.listdir(config.pdb_dir))

FileNotFoundError: [Errno 2] No such file or directory: '/vol/tmp/kormanav/pdb_dir/pdb6rew_copy.ent' -> '/vol/tmp/kormanav/pdb_dir/6rew_copy.pdb'
```
Whereby this also happens when the file is placed in the pdb_dir:
```
ls /vol/tmp/kormanav/pdb_dir
3eiy.pdb
6rew_copy.pdb	
6rew.pdb
```

**Expected behavior**
- I want to be able to use dssp features like rsa etc. on structures that can not be simply downloaded from the PDB database.
- No PDB download should be triggered when using paths to local pdb files.

**Desktop (please complete the following information):**

- OS: macOS
- Python Version: 3.8
- Graphein Version: Graphein on master branch (version 1.4.0), with pip install -e

### PR DESCRIPTION
**Pull request by avivko**
#### Reference Issues/PRs
- Fixes #171 (was not properly fixed by #172 ) 

#### What does this implement/fix? Explain your changes

- Includes refactoring of g.graph to always include the parameters "name", "pdb_code", and "pdb_path".
- adds hover titles to the correlation graphs, so that we know which residue we are looking at.

Rational for refactoring:
Before this PR, after we read a pdb and make a DataFrame out of it, as far as I know, we didn't save the original path to the pdb in the graph metadata. This is why previously, once we wanted to get the dssp features of a local pdb, we ran into problems as long as the structure was not in the pdb_dir. I wanted to make G.graph always contain the following three separate parameters from here on out: name, pdb_code, pdb_path. This way, when pdb_code is not None, we know we have a pdb from the protein database (and can also use download_pdb(pdb_code) etc.); Otherwise, we know are dealing with a local pdb and we keep the path to where it was originally stored. 
It is important to note that the pdb file might be moved after a graph was generated, therefore yielding a dead pdb_path.
For getting dssp features, this PR also provides the option to provide a new pdb_path in such a case.

#### What testing did you do to verify the changes in this PR?
I ran the Jupyter notebooks from the protein tutorials as well as my own local pdb + dssp graph conversion workflow and it all works.

### CODE REVIEW

File: graphein/protein/graphs.py [hunk 1]
```diff
@@ -69,8 +69,9 @@ def read_pdb_to_dataframe(
     :returns: ``pd.DataFrame`` containing protein structure
     :rtype: pd.DataFrame
     """
-    if pdb_code is None and pdb_path is None:
-        raise NameError("One of pdb_code or pdb_path must be specified!")
+
+    assert (pdb_code and not pdb_path) or (not pdb_code and pdb_path), 'Either a PDB ID or a path to a local PDB file' \
```
Comment (a-r-j) [inline graphein/protein/graphs.py:73]: Would you mind running `black` to format the quotes

File: graphein/protein/graphs.py [hunk 2]
```diff
@@ -359,17 +362,33 @@ def initialise_graph_with_metadata(
     :type protein_df: pd.DataFrame
     :param raw_pdb_df: Unprocessed dataframe of protein structure for comparison and traceability downstream.
     :type raw_pdb_df: pd.DataFrame
-    :param pdb_id: PDB Accession code.
-    :type pdb_id: str
     :param granularity: Granularity of the graph (eg ``"atom"``, ``"CA"``, ``"CB"`` etc or ``"centroid"``).
         See: :const:`~graphein.protein.config.GRAPH_ATOMS` and :const:`~graphein.protein.config.GRANULARITY_OPTS`.
     :type granularity: str
+    :param name: given name for graph
+    :type name: str
```
Comment (a-r-j) [inline graphein/protein/graphs.py:369]: Need optional annotations and default values

File: graphein/protein/graphs.py [hunk 3]
```diff
@@ -359,17 +362,33 @@ def initialise_graph_with_metadata(
...
+    :param pdb_code: PDB ID / Accession code, if the PDB is available on the PDB database.
+    :type pdb_code: str
+    :param pdb_path: path to local PDB file.
+    :type pdb_path: str
     :return: Returns initial protein structure graph with metadata.
     :rtype: nx.Graph
     """
+
+    assert (pdb_code and not pdb_path) or (not pdb_code and pdb_path), 'Either a PDB ID or a path to a local PDB file' \
```
Comment (a-r-j) [inline graphein/protein/graphs.py:378]: again, `black`

File: graphein/protein/graphs.py [hunk 4]
```diff
@@ -520,9 +540,11 @@ def construct_graph(
 
     :param config: :class:`~graphein.protein.config.ProteinGraphConfig` object. If None, defaults to config in ``graphein.protein.config``.
     :type config: graphein.protein.config.ProteinGraphConfig, optional
+    :param name: an optional given name for the graph. the PDB ID or PDB file name will be used if not specified.
+    :type name: str, optional
     :param pdb_path: Path to ``pdb_file`` to build graph from. Default is ``None``.
     :type pdb_path: str, optional
-    :param pdb_code: 4-character PDB accession pdb_code to build graph from. Default is ``None``.
+    :param pdb_code: 4-character PDB pdb ID / accession code  to build graph from. Default is ``None``.
```
Comment (a-r-j) [inline graphein/protein/graphs.py:547]: typo in docstring

File: graphein/protein/graphs.py [hunk 5]
```diff
@@ -540,14 +562,13 @@ def construct_graph(
     :type: nx.Graph
     """
 
+    assert (pdb_code and not pdb_path) or (not pdb_code and pdb_path), 'Either a PDB ID or a path to a local PDB file' \
```
Comment (a-r-j) [inline graphein/protein/graphs.py:565]: again, `black`

**Review by a-r-j — COMMENTED** [review]
LGTM! Would you mind checking out the failing notebook execution?

**Comment by avivko** [comment]
Would you like me to me to also add a test for dssp with both a pdb code a local pdb structure? 
I could also add the show_edges visualization from #167 to the corresponding notebook while I'm already on it.

**Comment by a-r-j** [comment]
I think a test & example of the edges in the asteroid plot would be great :)

**Review by a-r-j — APPROVED** [review]

**Pull request merged**